provider_empty.go

```diff
@@ -0,0 +1,12 @@
+package jwtauth
+
+// EmptyProvider is used mainly for testing, until real providers are added
+type EmptyProvider struct{}
```

Comment:
Should we bump this down to a sub-directory `providers` so that future providers have a place to be separate? As it stands if someone wanted to add a custom provider, I'm not sure if it would be clear to them where to put the file. What do you think?

Reply:
I was thinking each provider would be in its own separate file, i.e. `provider_azure.go`, `provider_gsuite.go`, etc. I think the reasoning for having it all in the same directory here was to avoid circular import problems.

Reply:
👍